user-service/controller/user-controller.js

```diff
@@ -34,15 +38,79 @@ export async function signIn(req, res) {
         const { username, password } = req.body;
         const user = await _FindOneuser(username,password)
         if (user) {
-            const accessToken = jwt.sign({username:username},ACCESS_TOKEN_SECRET)
-            console.log(accessToken)
-            // res.json({accessToken:accessToken})
-            res.cookie("accesstoken",accessToken,{httpOnly:true})
-            return res.status(200).json({message: `Log in is successful!`,accesstoken:accessToken });
+            const accessToken = jwt.sign({username:username},ACCESS_TOKEN_SECRET,{expiresIn:"1h"})
+            const refreshToken = jwt.sign({username:username},REFRESH_ACCESS_TOKEN_SECRET,{expiresIn:"1h"})
+            user.refreshToken = refreshToken
+            user.save()
+            //refresh token in cookie
+            //access token in data
+            if(validatePassword(password, user.password)){
+                return res.cookie("refreshtoken",refreshToken,{httpOnly:true}).status(200).json({message: `Log in is successful!`,accesstoken:accessToken });
+            }  else {
+                return res.status(400).json({message: 'Incorrect username/password!'});
+            }
+            
         } else {
             return res.status(400).json({message: 'Incorrect username/password!'});
         }
     } catch (err) {
         return res.status(500).json({message: 'Database failure when trying to log in!'})
     }
 }
+
+
+
+export async function logout(req, res) {
+    try {
+        // Get token from request
+        const accessToken = req.accessToken
+        if (!accessToken) {
+            return res.status(401).json({message: 'Token is missing or invalid!'});
+        }
+        const resp = await _addBlackList(accessToken);
+        if (resp.err) {
+            return res.status(409).json({message: 'Unable to add to blacklist'});

```

Comment:
Can I check why is the error response status 409? What is the conflict here? From what I read, 409 is generally used for PUT requests.

Reply:
You are right, 409 usually entails 2 http request having conflicts for the same resource. I think it may have been a typo on my side. Changed it to 500 instead which represents internal server error!